### ISSUE TITLE
Ignore on inline comments

### ISSUE THREAD

**Issue by zchee** [zchee/deoplete-go#39]
Currently, complements even during inline comments.
e.g. 

``` go
fmt.Printf("|")
//          ^
//          | cursor
```

I don’t mind either way.
Please let us know what you think.

![screenshot 2016-02-10 07 59 37](https://cloud.githubusercontent.com/assets/6366270/12933603/4de0164c-cfcc-11e5-8bc9-d056ed18f232.png)

**Comment by mmlb** [comment]
Yeah I've noticed this and have found it slightly annoying, it would be nice to not have it.

**Comment by Shougo** [comment]
Hm...
I can add the feature though.

**Comment by zchee** [comment]
@Shougo Its meaning is add implements to deoplete?

**Comment by Shougo** [comment]
neocomplete implements near feature: `neocomplete#helper#get_syn_name()`.

**Comment by zchee** [comment]
@Shougo Ok, I check it now.

**Comment by zchee** [comment]
@Shougo I understand.
Is it like this?

e.g. (sloppy...)

``` vim
let neocomplete.within_comment = neocomplete#helper#get_syn_name(1) =~? '"'
```

**Comment by Shougo** [comment]
Yes, but it is slow.
It should be called if it is really needed.

**Comment by zchee** [comment]
@Shougo Oh...

Tentatively, realized in this. based by https://github.com/zchee/deoplete-go/pull/19
In golang, Comments start is `//` or `/*`. 
It does not supported `/*` and buggy,
https://gist.github.com/zchee/72e2007991c188030742#file-deoplete-go-py-L48-L49

Is this also slow?

**Comment by fatih** [comment]
YCM has this option `g:ycm_complete_in_comments`: https://github.com/Valloric/YouCompleteMe/blob/master/doc/youcompleteme.txt#L1970-L1977 

Which is a setting to enable/disable inside `comments`

And `g:ycm_complete_in_strings`: https://github.com/Valloric/YouCompleteMe/blob/master/doc/youcompleteme.txt#L1980-L1994

For enabling/disabling inside `strings`

Would be great if could have it inside `deoplete` or somehow inside `deoplete-go`. Not sure which is the correct place.

**Comment by Shougo** [comment]
> Is this also slow?

It is not slow.  But it is too hard to parse the comment correctly.
The Vim syntax feature is slow but it is more correctly.

**Comment by Shougo** [comment]
@fatih I think YCM already implements same feature with neocomplete.
But I think it is slow.

**Comment by Shougo** [comment]
I have added `get_syn_name()` in deoplete.util.

**Comment by zchee** [comment]
@Shougo Thanks!
I'll check later.

> It is not slow. But it is too hard to parse the comment correctly.
> The Vim syntax feature is slow but it is more correctly.

I understand. sure, it seems difficult...

**Comment by Shougo** [comment]
I have implemented "disabled_syntaxes" feature.

**Comment by zchee** [comment]
@Shougo Thanks!!
Fix of the code of `deoplete-go` is no longer required. You have saved me.

@fatih Your wishes also came true in `deoplete.nvim` core implementation.

Also, this issue thread has been resolved, Close.
